addons/base_import/models/base_import.py

```diff
@@ -982,6 +1217,121 @@ def do(self, fields, columns, options, dryrun=False):
 
         return import_result
 
+    def _handle_multi_mapping(self, import_fields, input_file_data):
+        """ This method handles multiple mapping on the same field.
+        It will return the list of the mapped fields and the concatenated data for each field:
+            -> if two column are mapped on the same text or char field, they will end up
+            in only one column, concatenated via space (char) or new line (text).
+
+        Input/output Example:
+            input_file_data: [
+                ["Value part 1", "1", "res.partner_id1", "Value part 2"],
+                ["I am", "1", "res.partner_id1", "Batman"],
+            ]
+            fields: [desc, some_number, partner, desc]
+
+            output merged_data: [
+                ["Value part 1 Value part 2", "1", "res.partner_id1"],
+                ["I am Batman", "1", "res.partner_id1"],
+            ]
+            fields: [desc, some_number, partner]
+        """
+        # Get fields and their occurrences indexes
+        mapping_indexes = collections.defaultdict(list)
```

Comment:
Mmh. 
"file_data_indexes_by_field"?

Maybe a bit verbose.
Not a fan of "mapping_indexes" though.